components/support/nimbus-fml/src/defaults/validator.rs

```diff
@@ -222,6 +244,254 @@ impl<'a> DefaultsValidator<'a> {
+        // we want to skip validation on the `skip` type ref: this is only set by the property defining
+        // a string-alias.
+        let should_validate = |v: &TypeRef| -> bool { !matches!(skip, Some(t) if t == v) };
+        match (typ, value) {
+            (TypeRef::StringAlias(_), Value::String(s)) => {
+                check_string_aliased_property(path, error_path, typ, s, defn, errors)
+            }
+            (TypeRef::Option(_), &Value::Null) => (),
+            (TypeRef::Option(inner), _) => {
+                self.validate_string_aliases(path, error_path, inner, value, defn, skip, errors)
+            }
+            (TypeRef::List(inner), Value::Array(array)) => {
+                if should_validate(inner) {
+                    for value in array {
+                        self.validate_string_aliases(
+                            path, error_path, inner, value, defn, skip, errors,
+                        );
+                    }
+                }
+            }
+            (TypeRef::EnumMap(kt, vt), Value::Object(map)) => {
+                if should_validate(kt) && matches!(**kt, TypeRef::StringAlias(_)) {
+                    for value in map.keys() {
+                        check_string_aliased_property(path, error_path, kt, value, defn, errors);
+                    }
+                }
+
+                if should_validate(vt) {
+                    for (key, value) in map {
+                        let path = format!("{path}['{key}']");
+                        let error_path = append_quoted(error_path, key);
+
+                        self.validate_string_aliases(
+                            &path,
+                            &error_path,
+                            vt,
+                            value,
+                            defn,
+                            skip,
+                            errors,
+                        );
+                    }
+                }
+            }
+            (TypeRef::StringMap(vt), Value::Object(map)) => {
+                if should_validate(vt) {
+                    for (key, value) in map {
+                        let path = format!("{path}['{key}']");
+                        let error_path = append1(error_path, key);
+
+                        self.validate_string_aliases(
+                            &path,
+                            &error_path,
+                            vt,
+                            value,
+                            defn,
+                            skip,
+                            errors,
+                        );
+                    }
+                }
+            }
+            (TypeRef::Object(nm), Value::Object(map)) => {
+                let path = format!("{path}#{nm}");
+                let error_path = append1(error_path, "{");
+                let obj_def = self.get_object(nm).unwrap();
+
+                for prop in &obj_def.props {
+                    let nm = &prop.name;
```

Comment:
Can we call this name? I saw the `{nm}` format string below and I had to double check what it was.

Reply:
Rust conventions seem to be stuck in the Age of the Great Vowel Shortage, and re-using local variables names.